From: Jeff Mahoney <jeffm@suse.com>
Subject: [PATCH 02/40] reiserfs: don't use BUG when panicking

 This patch changes reiserfs_panic() to use panic() initially instead of
 BUG(). Using BUG() ignores the configurable panic behavior, so systems
 that should be failing and rebooting are left hanging. This causes problems
 in active/standby HA scenarios.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
---
 fs/reiserfs/prints.c |    7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

--- a/fs/reiserfs/prints.c	2007-07-14 12:43:04.000000000 -0400
+++ b/fs/reiserfs/prints.c	2007-08-27 14:03:37.000000000 -0400
@@ -356,13 +356,8 @@ extern struct tree_balance *cur_tb;
 void reiserfs_panic(struct super_block *sb, const char *fmt, ...)
 {
 	do_reiserfs_warning(fmt);
-	printk(KERN_EMERG "REISERFS: panic (device %s): %s\n",
+	panic(KERN_EMERG "REISERFS: panic (device %s): %s\n",
 	       reiserfs_bdevname(sb), error_buf);
-	BUG();
-
-	/* this is not actually called, but makes reiserfs_panic() "noreturn" */
-	panic("REISERFS: panic (device %s): %s\n",
-	      reiserfs_bdevname(sb), error_buf);
 }
 
 void reiserfs_abort(struct super_block *sb, int errno, const char *fmt, ...)
